test/e2e/dra/test-driver/app/kubeletplugin.go

```diff
@@ -162,10 +171,60 @@ func (ex *ExamplePlugin) IsRegistered() bool {
 	return status.PluginRegistered
 }
 
-// Block sets a flag to block Node[Un]PrepareResources
+// WithBlockedNodePrepareResources locks blockPrepareResourcesMutex to block NodePrepareResources
+// to emulate time consuming or stuck calls
+func (ex *ExamplePlugin) WithBlockedNodePrepareResources(f func() (*v1.Pod, func())) (*v1.Pod, func()) {
+	ex.blockPrepareResourcesMutex.Lock()
+	defer ex.blockPrepareResourcesMutex.Unlock()
+	return f()
+}
+
+// WithBlockedNodeUnprepareResources locks blockUnprepareResourcesMutex to block NodeUnprepareResources
 // to emulate time consuming or stuck calls
-func (ex *ExamplePlugin) Block() {
-	ex.block = true
+func (ex *ExamplePlugin) WithBlockedNodeUnprepareResources(f func() (*v1.Pod, func())) (*v1.Pod, func()) {
+	ex.blockUnprepareResourcesMutex.Lock()
+	defer ex.blockUnprepareResourcesMutex.Unlock()
+	return f()
+}
+
+func (ex *ExamplePlugin) WithNodePrepareResourcesFailure(f func() (*v1.Pod, func())) (*v1.Pod, func()) {
+	ex.failPrepareResourcesMutex.Lock()
+	ex.prepareResourcesFailure = errors.New("simulated PrepareResources failure")
+	ex.failPrepareResourcesMutex.Unlock()
+
+	defer func() {
+		ex.failPrepareResourcesMutex.Lock()
+		ex.prepareResourcesFailure = nil
+		ex.failPrepareResourcesMutex.Unlock()
+	}()
+
+	return f()
+}
+
+func (ex *ExamplePlugin) GetPrepareResourcesFailure() error {
```

Comment:
This doesn't need to be exported? If it did, it should have been documented.

Reply:
done